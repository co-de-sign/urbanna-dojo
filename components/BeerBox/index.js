import React, { Component } from 'react'
import { StyleSheet, View, Text } from 'react-native'

export default class BeerBox extends Component {
  render() {
    const { data } = this.props

    return (
      <View style={styles.box}>
        <Text>{data.name}</Text>
        <Text>{data.alcohol}% alcool</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  box: {
    marginTop: 32,
    padding: 24,
    height: 224,
    width: 256,
    elevation: 1,
    backgroundColor: '#fff',
  }
})
