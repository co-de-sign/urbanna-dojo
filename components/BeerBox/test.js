import React from 'react';
import renderer from 'react-test-renderer';

import BeerBox from '.'

describe('BeerBox', () => {
  let box

  beforeEach(() => {
    const boxData = {
      name: 'Fake name',
      alcohol: .5,
    }

    box = renderer.create(<BeerBox data={boxData} />)
  })

  it('renders correctly', () => {
    expect(box).toMatchSnapshot();
  })
})
