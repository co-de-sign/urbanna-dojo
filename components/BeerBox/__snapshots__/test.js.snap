// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerBox renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "elevation": 1,
      "height": 224,
      "marginTop": 32,
      "padding": 24,
      "width": 256,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Fake name
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    0.5
    % alcool
  </Text>
</View>
`;
